node/src/components/linear_chain/state.rs

```diff
@@ -114,19 +136,10 @@ impl LinearChain {
             ..
         } = fs.clone();
         if let Some(latest_block) = self.latest_block.as_ref() {
-            // If it's a switch block it has already forgotten its own era's validators,
-            // unbonded some old validators, and determined new ones. In that case, we
-            // should add 1 to last_block_era.
-            let current_era = latest_block.header().era_id()
-                + if latest_block.header().is_switch_block() {
-                    1
-                } else {
-                    0
-                };
```

Comment:
Are we sure it's safe to remove this? I can't say I remember why it was needed, but I vaguely recall that it was fixing some kind of an edge case...

Reply:
I believe it's equivalent: [`next_block_era_id` does the switch block check internally.](https://github.com/casper-network/casper-node/blob/00451e82dd8d5cabdb24567464925c9e1c02becc/node/src/types/block.rs#L730)

Reply:
Ah, I missed that you changed `era_id` to `next_block_era_id`. Yes, it should be equivalent, then :+1: